descartes_planner/include/descartes_planner/ladder_graph.h

```diff
@@ -0,0 +1,224 @@
+#ifndef DESCARTES_LADDER_GRAPH_H
+#define DESCARTES_LADDER_GRAPH_H
+
+#include "descartes_core/trajectory_id.h"
+#include "descartes_core/trajectory_timing_constraint.h"
+
+namespace descartes_planner
+{
+
+struct Rung
+{
+  descartes_core::TrajectoryID id; // corresponds to user's input ID
+  descartes_core::TimingConstraint timing; // user input timing
+  std::vector<double> data; // joint values stored in one contiguous array
+};
+
+struct __attribute__ ((__packed__)) Edge
+{
+  double cost; // transition cost from vertex who owns this object to 'idx' in next rung
+  unsigned idx; // from THIS rung to 'idx' into the NEXT rung
+};
+
+/**
+ * @brief LadderGraph is an adjacency list based, directed graph structure with vertices
+ *        arranged into "rungs" which have connections only to vertices in the adjacent
+ *        rungs. Assumes a fixed DOF.
+ */
+class LadderGraph
+{
+public:
+  using size_type = std::size_t;
```

Comment:
As far as the language is concerned there isn't a difference.

The new syntax is just a little nicer especially in cases like:
```c++
using fp = void (*)(const& Foo);
```
when the alternative is:
```c++
typedef void (*fp)(const& Foo);
```

Not used here, but you can combine templates with `using` as well:
```c++
template<typename T>
using Vec = std::vector<T>;

Vec<Foo> bar;
```